### PR TITLE
br : migrate test-infra to testify for lightning/backend/backend_test

### LINKED ISSUE #29983: migrate test-infra to testify for `lightning/backend/backend_test.go`

### PR DESCRIPTION
**Pull request by JayL-zxl**
### What problem does this PR solve?

Issue Number: close  #29983

Problem Summary:

### What is changed and how it works?
migrate test-infra to testify for lightning/backend/backend_test.go
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by lcwangchao — APPROVED** [review]

File: br/pkg/lightning/backend/backend_test.go [hunk 1]
```diff
@@ -31,22 +32,25 @@ func Test(t *testing.T) {
...
 	s.controller.Finish()
 }
 
-func (s *backendSuite) TestOpenCloseImportCleanUpEngine(c *C) {
-	s.setUpTest(c)
+func TestOpenCloseImportCleanUpEngine(t *testing.T) {
+	s := setUpTest(t)
 	defer s.tearDownTest()
-
+	setUpTest(t)
```
Comment (tisonkun) [inline br/pkg/lightning/backend/backend_test.go:53]: Why do you call `setUpTest` twice in one test case?

Besides, please consider whether we can parallelize these tests.

Comment (JayL-zxl) [reply br/pkg/lightning/backend/backend_test.go:53]: sorry ..is a mistake

Comment (JayL-zxl) [reply br/pkg/lightning/backend/backend_test.go:53]: parallelize is okay

File: br/pkg/lightning/backend/backend_test.go [hunk 2]
```diff
@@ -31,22 +32,25 @@ func Test(t *testing.T) {
 
 // FIXME: Cannot use the real SetUpTest/TearDownTest to set up the mock
 // otherwise the mock error will be ignored.
```
Comment (tisonkun) [inline br/pkg/lightning/backend/backend_test.go:34]: This comment is no longer valid.

File: br/pkg/lightning/backend/backend_test.go [hunk 3]
```diff
@@ -31,22 +32,25 @@ func Test(t *testing.T) {
...
-
-func (s *backendSuite) setUpTest(c gomock.TestReporter) {
-	s.controller = gomock.NewController(c)
-	s.mockBackend = mock.NewMockBackend(s.controller)
-	s.backend = backend.MakeBackend(s.mockBackend)
-	s.ts = oracle.ComposeTS(time.Now().Unix()*1000, 0)
+func setUpTest(c gomock.TestReporter) *backendSuite {
```
Comment (tisonkun) [inline br/pkg/lightning/backend/backend_test.go:35]:
```suggestion
func createBackendSuite(c gomock.TestReporter) *backendSuite {
```

Also please remove redundant code snippet above:

```go
var _ = Suite(&backendSuite{})
func Test(t *testing.T) {
	TestingT(t)
}
```

Make sure you understand how tests are running.

Comment (JayL-zxl) [reply br/pkg/lightning/backend/backend_test.go:35]: sorry. i forget to delete it. is my mistake.

File: br/pkg/lightning/backend/backend_test.go [hunk 4]
```diff
@@ -189,11 +189,12 @@ func (s *backendSuite) TestOpenEngineFailed(c *C) {
 		Return(errors.New("fake unrecoverable open error"))
 
 	_, err := s.backend.OpenEngine(ctx, &backend.EngineConfig{}, "`db`.`table`", 1)
-	c.Assert(err, ErrorMatches, "fake unrecoverable open error")
+	require.Equal(t, "fake unrecoverable open error", err.Error())
```
Comment (tisonkun) [inline br/pkg/lightning/backend/backend_test.go:192]:
```suggestion
	require.EqualError(t, err, "fake unrecoverable open error")
```

**Review by tisonkun — COMMENTED** [review]
LGTM. Let me push one more tiny change and wait for CI gives green status.

**Comment by tisonkun** [comment]
check_dev_2 failed on `executor` pkg.

/merge

**Review by tisonkun — APPROVED** [review]

**Pull request merged**